zebra-state/src/service/non_finalized_state.rs

```diff
@@ -92,24 +105,41 @@ impl NonFinalizedState {
             .or_else(|| {
                 self.chain_set
                     .iter()
-                    .find_map(|chain| chain.fork(parent_hash))
+                    .find_map(|chain| {
+                        let mut mmr = finalized_tip_mmr.clone();
+                        // TODO: pass Sapling and Orchard roots for each block. How?
```

Comment:
Depends on #1320, check with @dconnolly how that is going to work

Reply:
I've shuffled things around, now I realize I will probably get them from `PreparedBlock`

Reply:
Sounds good to me!